plugins/include/amxconst.inc

```diff
@@ -55,10 +55,25 @@ public stock const NULL_STRING[1];
 public stock const Float:NULL_VECTOR[3];
 
 /**
- * The maximum buffer size required to store a clients name.
+ * The maximum buffer size required to store a client's name.
  */
 #define MAX_NAME_LENGTH 32
 
+/**
+ * The maximum buffer size required to store a client's IP address without a port.
+ */
+#define MAX_IP_LENGTH 16
+
+/**
+ * The maximum buffer size required to store a client's IP address without a port.
+ */
+#define MAX_IP_WITH_PORT_LENGTH 22
+
+/**
+ * The maximum buffer size required to store a client's AuthID.
+ */
+#define MAX_AUTHID_LENGTH 34
```

Comment:
Yeah, I set it to 34 according to that as well. I've been using 35 in my plugins though.

Reply:
The engine uses 64. I would use the same vlaue.

Reply:
I changed it.

Reply:
Looks like now I understand why it is 34. 34 is 33 + 1 for `EOS` of course. Length of `STEAM_12345:1234567890:1234567890` is exactly 33. [Look](https://github.com/dreamstalker/rehlds/blob/469908f5c5e9148bf936610c696b8b4ee87c3634/rehlds/engine/sv_main.cpp#L7952-L7953):
```Pawn
TSteamGlobalUserID steam2ID = Steam_Steam3IDtoSteam2(id->m_SteamID);
Q_snprintf(idstr, ARRAYSIZE(idstr) - 1, "STEAM_%u:%u:%u", steam2ID.m_SteamInstanceID, steam2ID.m_SteamLocalUserID.Split.High32bits, steam2ID.m_SteamLocalUserID.Split.Low32bits);
```
A part is `m_SteamInstanceID` which [is defined as `SteamInstanceID_t`](https://github.com/dreamstalker/rehlds/blob/65c6ce593b5eabf13e92b03352e4b429d0d797b0/rehlds/common/SteamCommon.h#L691) (an [alias](https://github.com/dreamstalker/rehlds/blob/65c6ce593b5eabf13e92b03352e4b429d0d797b0/rehlds/common/SteamCommon.h#L662) for `unsigned short` and there is a note that it should be 16 bits), so it should be 5 digits max.
B and C parts are `High32bits` and `Low32bits`, which size in bits is obvious from their names, so they're both 10 digits max.

I'm not sure why the engine uses 64, but I think this is for simplicity (or for simpler code modification in future?). And also I'm not sure if we can rely on these implementation details, but the engine relies on these steam struct fields. Moreover [it is hardcoded](https://github.com/dreamstalker/rehlds/blob/65c6ce593b5eabf13e92b03352e4b429d0d797b0/rehlds/public/steam/steamclientpublic.h#L547-L555) (this code is also compiled into the engine!) that `m_SteamInstanceID` is always `0`, `High32bits` is always 1-bit and `Low32bits` is always even. But there is also [`m_unAccountInstance`](https://github.com/dreamstalker/rehlds/blob/65c6ce593b5eabf13e92b03352e4b429d0d797b0/rehlds/public/steam/steamclientpublic.h#L760) which is even 20-bit! Valve in future can just update the steam with the game engines and extend this format. So... if 64 in the engine code is for future modifications, then it would be better to stick with it.